converter/src/test/java/gov/cms/qpp/acceptance/cpc/CpcPlusAcceptanceTest.java

```diff
@@ -60,18 +72,64 @@ void testCpcPlusFileFailures() throws IOException {
 		List<Path> successesThatShouldBeErrors = getXml(FAILURE)
 			.filter(entry -> {
 				Converter converter = new Converter(new PathQrdaSource(entry));
-
+				//TODO remove this once all error scenarios are validated
+				if (fixtureValues.get(entry.toFile().getName()) == null) {
+					return false;
+				}
 				try {
 					converter.transform();
 					return true;
 				} catch (TransformException expected) {
+					//runnning conversions on individual files
+					List<Detail> details = expected.getDetails().getErrors().get(0).getDetails();
+					verifyOutcome(entry.toFile().getName(), details);
 					return false;
 				}
 			}).collect(Collectors.toList());
 
+		assertWithMessage("The fixture file is not representative of the failures directory")
+				.that(getXml(FAILURE).count())
+				.isEqualTo(fixtureValues.size());
 		assertThat(successesThatShouldBeErrors).isEmpty();
 	}
 
+	private void verifyOutcome(String filename, List<Detail> details) {
+		CPCAcceptanceFixture expectedErrors = fixtureValues.get(filename);
+
+		System.out.println("Verifying scenario " + filename +
+				", expected errors match the following actual errors. " + details);
+
+		if (expectedErrors.isStrict()) {
```

Comment:
Don't we always want to be strict about this?  I saw in the meta-data that we are not strict in some scenarios.

Reply:
We don't want to always be strict.  In the future, other validations may come along that makes one of these CPC+ XML files generate a new validation error and that would then throw off the strict assert and fail the test.  _But_, we shouldn't need to care that this XML file now has a new additional validation error on it.  We should only care that it still fails in the original way.

Reply:
But is it a bad thing to always be strict?  It ensures that we've enumerated all the errors that could occur with the file.  If a new validation causes a file to throw an error, then add it to the list of expected validation errors.  That way we know everything that's expected and that there's no issues with unexpected interactions.

Reply:
I think it is bad thing, actually.  These CPC+ files were added to test the existence of specific errors in them, so we should only test for those specific errors and not any possible additional ones.

In past projects I have been on, we decided to be strict.  This led to many tests failing when small changes were introduced.  At best, this creates unnecessary churn and a groaning engineer, and at worst, creates an environment where the engineer is overwhelmed with all the test failures and just decides to fix the test by grabbing the actual results and making them the expected results (and ta-da, the test passes now).